Yoast/wordpress-seo
# Make it possible to start the configuration wizard from a specific step

## Summary

This PR can be summarized in the following changelog entry:

* [not user-facing] Make it possible to start the configuration wizard from a specific step.

## Relevant technical choices:

* I've changed the configuration step identifiers to make them more URL-friendly. For example: `titleTemplate` became `title-template`.

## Test instructions

This PR can be tested by following these steps:

* Link this the `12158-add-fragments-for-configuration-wizard` from yoast-components to this branch.
* Go to `[localdomain]/wp-admin/admin.php?page=wpseo_configurator#newsletter` for example, `http://local.wordpress.test/wp-admin/admin.php?page=wpseo_configurator#newsletter`.
* See that you go directly to the newsletter step from the configuration wizard.
* Click on some steps at the top of the screen. 
* See that the hash in the URL changes.
* Click on the 'previous' and 'next' button in the lower right corner.
* See that they still work as expected, and the hash in the URL changes.
* Use your browser's 'previous' and 'next' button.
* See that they still work as expected, and the hash in the URL changes.
* Go to a non-existing has, for example `http://local.wordpress.test/wp-admin/admin.php?page=wpseo_configurator#bogus`.
* See that you end up on the intro page of the wizard, and that the hash in the url says #intro.
* Because I changed the step identifiers (see 'Relevant technical choices'), make sure to visit every step of the wizard and check if nothing disappeared and no console errors are thrown.
* Test the same steps in the stand-alone by running `yarn start` in `yoast-components` and going to `localhost:3333`. Only the direct navigation to for example `http://localhost:3333/#multipleAuthors` doesn't work in the stand-alone.

## UI changes

## Quality assurance


Fixes #12158
